spec/controllers/hq/sponsorships_controller_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe Hq::SponsorshipsController, type: :controller do
+  let(:sponsorship) {create :sponsorship, active: true}
+
+  before :each do
+    sign_in instance_double(AdminUser)
+  end
+
+  specify "#inactivate" do
+    expect(Sponsorship).to receive(:find).and_return(sponsorship)
+    put :inactivate, id: sponsorship.id, sponsor_id: sponsorship.sponsor_id, sponsorship: {end_date: Date.current}
+
+    expect(sponsorship.active).to be false
```

Comment:
if we do `expect(sponsorship).to receive(:update!).and_return(true)` it will only return `true`, it will not change the `sponsorship.active`.   
Notice it is the same story we had with mocking the `.save`